tests/test_helpers.py

```diff
@@ -515,6 +516,106 @@ def test_proxies_from_env_http_with_auth(mocker) -> None:
     assert proxy_auth.encoding == "latin1"
 
 
+# --------------------- get_env_proxy_for_url ------------------------------
+
+
+@pytest.mark.parametrize(
+    ("proxies", "url_input", "expected_err_msg"),
+    (
+        (
+            {"no_proxy": ["aiohttp.io"], "NO_PROXY": ["aiohttp.io"]},
+            "http://aiohttp.io/path",
+            r"Proxying is disallowed for `'aiohttp.io'`",
+        ),
+        (
+            {
+                "http_proxy": ["http://example.com"],
+                "HTTP_PROXY": ["http://example.com"],
```

Comment:
I don't understand why it's necessary to have both spellings in the same test. It could be reasonable to test them separately, though. And maybe have different spellings with different values testing which one overrides another.

WDYT?

Reply:
You mean the case difference? If I remember correctly (been awhile, I'll go check) but there's no spec. for the proxy info, it's just by convention. And there's more than one; one with lower case and the other upper. I think there's even `all_proxy` `ALL_PROXY`. I looked at the source of [liburl.request.get_env_proxy_for_url](https://github.com/python/cpython/blob/7cb033c423b65def1632d6c3c747111543b342a2/Lib/urllib/request.py#L2500-L2503) and they just lower case it. So it doesn't matter, I guess I'm just used to seeing them together all the time.

Reply:
Yes, I mean the case difference. When you write a test, you communicate the expectations of how something is supposed to be used and to work in general. Point is that when you set both `http_proxy` **and** `HTTP_PROXY`, this creates a false impression (for any third party reader) and in order for this feature to work, the end-user is supposed to set two env variables with the same value. The underlying implementation you linked implies that the upper-case variant is always preferred and the lower-case is used as a fall-back.
The only reasonable case I see for setting both variables in a test is when the values are different: for example, a test could demonstrate that with `http_proxy=http://example.org HTTP_PROXY=http://example.com` set it would choose `http://example.com`.

Reply:
Make sense. Just doing a little debugging now. Before when I ran the tests I didn't have any environment variables set for the proxies, setting them, then running the tests, the tests fail. Seems like [monkeypatch.delenv(schema, False)](https://github.com/aio-libs/aiohttp/pull/4445/files/e717b2b9cc0ea5b5f3fa3d1a8c01f2a25d1e7cba#diff-08a6944653efc734f4c8cb856fc854b9f21d655ca69796af2b170c53bff7919eR608) is not working as I expected.